maplibre_gl/android/src/main/java/org/maplibre/maplibregl/MapSnapshotWrapper.java

```diff
@@ -0,0 +1,181 @@
+package org.maplibre.maplibregl;
+
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
+import android.graphics.Canvas;
+import android.graphics.Color;
+import android.graphics.Paint;
+import android.graphics.PointF;
+import android.util.Log;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import io.flutter.plugin.common.MethodChannel;
+import org.maplibre.android.snapshotter.MapSnapshot;
+import org.maplibre.android.snapshotter.MapSnapshotter;
+import org.maplibre.android.camera.CameraPosition;
+import org.maplibre.android.geometry.LatLng;
+import org.maplibre.android.maps.MapLibreMap;
+import org.maplibre.android.maps.Style;
+import org.maplibre.android.style.layers.PropertyValue;
+
+import java.io.ByteArrayOutputStream;
+import java.util.List;
+import java.util.Map;
+
+public class MapSnapshotWrapper {
+    private static final String TAG = MapSnapshotWrapper.class.getSimpleName();
+    
+    private final MethodChannel channel;
+    private final android.content.Context context;
+    private MapSnapshotter snapshotter;
+    private MethodChannel.Result result;
+    
+    public MapSnapshotWrapper(MethodChannel channel, android.content.Context context) {
+        this.channel = channel;
+        this.context = context;
+    }
+    
+    public void startSnapshot(Map<String, Object> arguments, MethodChannel.Result result) {
+        this.result = result;
+        
+        try {
+            // Parse arguments
+            Integer width = (Integer) arguments.get("width");
+            Integer height = (Integer) arguments.get("height");
+            String styleUrl = (String) arguments.get("styleUrl");
+            Map<String, Object> cameraPositionMap = (Map<String, Object>) arguments.get("cameraPosition");
+            
+            if (width == null || height == null || styleUrl == null || cameraPositionMap == null) {
+                result.error("INVALID_ARGUMENTS", "Missing required arguments", null);
+                return;
+            }
```

Comment:
Also check if `styleUrl` is empty and, if so, return an error
```suggestion
            if (width == null || height == null || styleUrl == null || cameraPositionMap == null) {
                result.error("INVALID_ARGUMENTS", "Missing required arguments", null);
                return;
            }

            if (styleUrl.isEmpty()) {
                result.error("INVALID_ARGUMENTS", "Style URL cannot be empty", null);
                return;
            }
```